### PR TITLE
authorization page

### PR DESCRIPTION
**Pull request by NadiaMaksymchuk**
Task: [As a user I want to have authorization page](https://trello.com/c/5giQcBSP/4-as-a-user-i-want-to-have-authorization-page)

Responcive
image

Sign-up page
image

Sign-up validation
image

Sign-in page
image

Sign-in validation
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,96 +36,20 @@ _Tip: If you want to connect to the specific service outside of docker, then use
...
+      nvarchar Email
+      nvarchar PhoneCode
+      nvarchar PhoneNumber
+      nvarchar ImagePath
+      int Country
+      int Language
+      int TimeFormat
+      int DateFormat
+      boolean IsBanned
+      boolean IsDeleted
```
Comment (Anatolii-Ivanov) [inline README.md:51]: Please, check what you commit) Sadly, this is for a lot of files, you make merging development incorrectly...

File: frontend/package.json [hunk 2]
```diff
@@ -1,65 +1,65 @@
...
+        "karma": "^6.4.0",
+        "karma-chrome-launcher": "^3.1.1",
+        "karma-coverage": "^2.1.1",
+        "karma-jasmine": "^4.0.2",
+        "karma-jasmine-html-reporter": "~1.7.0",
+        "prettier": "^2.7.1",
+        "prettier-eslint": "^15.0.1",
+        "typescript": "^4.9.5"
+    }
 }
```
Comment (Anatolii-Ivanov) [inline frontend/package.json:65]: Let's stay with 2 symbols tabs here

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        <div class="dog-background"></div>
+        <img class="dog" src="assets/topic-icons/Dog Face.svg" />
+
+        <div class="monocle-face-background"></div>
+        <img class="monocle-face" src="assets/topic-icons/Face With Monocle.svg" />
+        <h1>
+            Learn English
+            <div>With Logo</div>
+        </h1>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-in/sign-in.component.html:53]: Can you move it to separate component? Not to repeat it)

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<div class="sign-up">
+    <form name="authForm" #authForm="ngForm" class="registration-form">
+        <h1>Sign In</h1>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-in/sign-in.component.html:3]: Ideally to make it as a general component, to which, based on route, we will set either login or sign up form

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        <label>
+            <mat-error *ngIf="submitted">{{ getEmailErrorMessage() }}</mat-error>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-in/sign-in.component.html:5]: Let's make error under input everywhere

File: frontend/src/app/modules/auth/sign-in/sign-in.component.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    getEmailErrorMessage() {
+        if (this.form.controls['email'].hasError('required')) {
+            return 'You must enter a value';
+        }
+
+        if (this.form.controls['email'].hasError('email')) {
+            return 'Not a valid email';
+        }
+
+        return this.form.controls['email'].hasError('maxlength') ? 'Can not be more than 30 characters' : '';
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-in/sign-in.component.ts:32]: Hmmm, what about creating a function and dictionary of validation rules?)
{ 'required', 'You muse enter a value' }
And just make the same

File: frontend/src/app/modules/auth/sign-up/sign-up.component.sass [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        font-size: 330%
+        line-height: 127.5%
+        text-align: center
+        text-shadow: -2px 2px 1px $accent-color
+    .car-background
+        @include background-icon
+        right: 30%
+        top: 46px
+    .car
+        position: absolute
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-up/sign-up.component.sass:66]: Have you tried to change width of your browser window?

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    getEmailErrorMessage() {
+        if (this.form.controls['email'].hasError('required')) {
+            return 'You must enter a value';
+        }
+
+        if (this.form.controls['email'].hasError('email')) {
+            return 'Not a valid email';
+        }
+
+        return this.form.controls['email'].hasError('maxlength') ? 'Can not be more than 30 characters' : '';
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-up/sign-up.component.ts:54]: The same here

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+                this.password,
+                [
+                    Validators.required,
+                    Validators.minLength(6),
+                    Validators.maxLength(25),
+                    Validators.pattern('^(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\\d@$!%*?&\\.]{6,}$'),
+                ],
+            ],
+            confirmPassword: [this.confirmPassword, Validators.required],
+        });
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-up/sign-up.component.ts:113]: Can't you ad it onInit? Won't it work only after form is dirty and submitted?

File: frontend/src/styles/colors.sass [hunk 10]
```diff
@@ -7,5 +7,3 @@ $gray-200: #CBC9D0
 $inactive-color: #E0E2F2
 $gray-100: #F6F5FA
 $white: #FFFFFF
-$selected-color: #61D4F6
-$button-color: #1C2FBA
```
Comment (Anatolii-Ivanov) [inline frontend/src/styles/colors.sass:11]: I guess it's needed)

File: frontend/tsconfig.json [hunk 11]
```diff
@@ -2,42 +2,30 @@
...
+        "importHelpers": true,
+        "target": "es2020",
+        "module": "es2020",
+        "lib": ["es2020", "dom"],
+        "paths": {
+            "@env/*": ["src/environments/*"],
+            "@core/*": ["src/app/core/*"],
+            "@shared/*": ["src/app/shared/*"],
+            "@modules/*": ["src/app/modules/*"]
+        }
```
Comment (Anatolii-Ivanov) [inline frontend/tsconfig.json:28]: And let's leave it as it was...

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]
Good work, I like your code, just some logic comments :)

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts (hunk 9 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-up/sign-up.component.ts:121]: I tried this way at first, but it should accept the value and then validate, and not during initialization

File: frontend/src/app/modules/auth/sign-up/sign-up.component.sass (hunk 7 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-up/sign-up.component.sass]: Of course, there is a screenshot in the description

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html (hunk 3 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-in/sign-in.component.html]: fixed

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html (hunk 4 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-in/sign-in.component.html]: fixed

File: frontend/src/app/modules/auth/sign-in/sign-in.component.html (hunk 5 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-in/sign-in.component.html]: fixed

File: frontend/src/app/modules/auth/sign-in/sign-in.component.ts (hunk 6 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-in/sign-in.component.ts]: got it, fixed

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts (hunk 8 above)
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-up/sign-up.component.ts]: fixed

File: frontend/tsconfig.json (hunk 11 above)
Comment (NadiaMaksymchuk) [reply frontend/tsconfig.json]: ok, fixed

File: frontend/src/styles/colors.sass (hunk 10 above)
Comment (NadiaMaksymchuk) [reply frontend/src/styles/colors.sass]: fixed

File: frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html
[the remark this answers is not in the document]
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html:1]: fixed

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+        lastName: new FormControl(this.lastName),
+        email: new FormControl(this.email),
+        password: new FormControl(this.password),
+        confirmPassword: new FormControl(this.confirmPassword),
+    });
+
+    constructor(private formBuilder: FormBuilder) {}
+
+    getEmailErrorMessage() {
+        if (this.form.controls['email'].hasError('required')) {
```
[the remark this answers is not in the document]
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/auth/sign-up/sign-up.component.ts]: Done

File: frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html:1]: Please, get back these files

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html
Comment (NadiaMaksymchuk) [reply frontend/src/app/modules/user-profile/select-topics-page/select-topics-page.component.html:1]: Done

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by bochka123 — APPROVED** [review]

**Pull request merged**